x-pack/platform/packages/shared/kbn-classic-stream-flyout/src/components/create_classic_stream_flyout/steps/name_and_confirm/name_and_confirm_step.tsx

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiFlexGroup } from '@elastic/eui';
+import { NameStreamSection } from './name_stream_section';
+import type { NameAndConfirmStepProps } from './types';
+
+export const NameAndConfirmStep = ({
+  template,
+  selectedIndexPattern,
+  onIndexPatternChange,
+  onStreamNameChange,
+  validationError = null,
+  conflictingIndexPattern,
+}: NameAndConfirmStepProps) => {
+  const indexPatterns = template.indexPatterns ?? [];
+
+  return (
+    <EuiFlexGroup data-test-subj="nameAndConfirmStep" direction="column" gutterSize="none">
+      <NameStreamSection
+        indexPatterns={indexPatterns}
+        selectedIndexPattern={selectedIndexPattern}
+        onIndexPatternChange={onIndexPatternChange}
+        onStreamNameChange={onStreamNameChange}
+        validationError={validationError}
+        conflictingIndexPattern={conflictingIndexPattern}
+      />
+    </EuiFlexGroup>
+  );
+};
```

Comment:
I didn't change this in my PR suggestion either because it would make the rewrite even more involved than it already is but still for future and general knowledge there is this concept that I really like 
https://julesblom.com/writing/parents-owners-data-flow (I would advice to read all of jules articles, every single one is a golden nugget)

So when you read it I think it will become clear that by introducing these tiny proxy components they only make the prop drill path longer and don't really introduce anything useful.

Read and  let me know if you agree and wdyt about the general philosophy.